### PR TITLE
Update MySQL connector to 8.0.18

### PR DESCRIPTION
**Pull request by tdcmeehan**
Depends on https://github.com/prestodb/testing-mysql-server/pull/9

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java [hunk 1]
```diff
@@ -161,7 +162,7 @@ else if (DATE.equals(type)) {
             // convert to midnight in default time zone
             long utcMillis = DAYS.toMillis(type.getLong(block, position));
             long localMillis = getInstanceUTC().getZone().getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
-            statement.setDate(parameter, new Date(localMillis));
+            statement.setDate(parameter, new Date(localMillis), Calendar.getInstance());
```
Comment (tdcmeehan) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java]: This is based on discussion found here: https://bugs.mysql.com/bug.php?id=91112

**Comment by tdcmeehan** [comment]
We need to make sure this works ok with prior versions of MySQL (v5.x for example)

**Comment by tooptoop4** [comment]
@tdcmeehan gentle ping

File: presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java [hunk 2]
```diff
@@ -28,8 +28,8 @@
 import com.facebook.presto.spi.type.Type;
 import com.facebook.presto.spi.type.VarcharType;
 import com.google.common.collect.ImmutableSet;
+import com.mysql.cj.jdbc.JdbcStatement;
 import com.mysql.jdbc.Driver;
```
Comment (ericsun2) [inline presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java]: should this be also changed to `com.mysql.cj.jdbc.Driver` according to https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-api-changes.html

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java (hunk 1 above)
Comment (ericsun2) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java]: Can we also make all the test JDBC URLs (e.g. presto-product-tests/conf/presto/etc/catalog/mysql.properties) look like `jdbc:mysql://mysql:13306?useSSL=false&serverTimezone=UTC` please?

Most 8.0+ instance requires useSSL=true|false, and serverTimezone=??? can address the above Date issue as well.

File: pom.xml [hunk 3]
```diff
@@ -759,7 +759,7 @@
             <dependency>
                 <groupId>mysql</groupId>
                 <artifactId>mysql-connector-java</artifactId>
-                <version>5.1.46</version>
+                <version>8.0.15</version>
```
Comment (ericsun2) [inline pom.xml]: Since 8.0.18 has been released, when we finally merge this PR, can we bump it up to 8.0.18 please?

**Comment by tdcmeehan** [comment]
Picking this back up.  Discovered a dependency that needs to be updated: prestodb/airlift#8

**Comment by ericsun2** [comment]
@tdcmeehan 
Can you kindly resolve the checks issue and get this PR merged please?

**Comment by ericsun2** [comment]
@tdcmeehan
The CI issue just needs a rebase, can you kindly get this PR going please?

**Pull request closed without merging**